tests/test_rewrite_visitor.py

```diff
@@ -96,25 +134,44 @@ def test_boolean_visitor_hard(self):
                 BooleanNot(BooleanVariable('x')), 
                 BooleanXor(
                     BooleanVariable('y'), 
-                    BooleanNot(BooleanConstant('c0')))))
+                    BooleanNot(
+                        BooleanConstant('c0')
+                    )
+                )
+            )
+        )
```

Comment:
This indentation is fine for me, but you forgot to have the same change in `constant_visitor`. It takes a bit more space that's why I prefer the other for the tests, which I also find more easily readable since it is not more compact by grouping the unary operators.